feature_engine/selection/target_mean_encoding.py

```diff
@@ -0,0 +1,264 @@
+
+    def fit(self, X, y, quantiles=5, test_size=0.3, random_state=0):
+        """
+        Performs
+        --------
+
+            1. Handling missing values and check invalid input variables.
+            2. Split data into train and test.
+            3. Performs the target mean encoding.
+            4. Update the results X_train_enc and X_test_enc.
+        
+        Returns
+        -------
+            None
+
+        Parameters
+        ----------
+
+            X: pandas dataframe of shape = [n_samples, n_features]
+                The input dataframe.
+
+            y: None
+                y is not needed for this transformer. You can pass y or None.
+            
+            test_size: float, default=0.3
+                The test size setting of the data in the train_test_split method.
+
+            random_state: int, default=0
+                The random state setting in the train_test_split method.
+
+            quantiles: int, default=5
+                The amount of quantiles setting in the qcut of the fit method in case of the numerical types.
+                The range of the quantiles values is in between 3 and 100. The transformer will return a message
+                if the input quantiles value is out of this range.
+
+        Attributes
+        ----------
+
+            X_test_enc: pandas dataframe of shape = [n_samples, n_features]
+                The output of target mean encoding of the test set.
+        
+        """
+
+        # check the target
+        if y is None:
+            raise ValueError('Please provide a target y for this encoding method.')
+
+        # check input dataframe
+        X = _is_dataframe(X)
+
+        # find all variables or check those entered are in the dataframe
+        self.variables = _find_all_variables(X, self.variables)
+        
+        # find categorical variables or check that those vars entered by the user
+        # are of type object
+        cat_variables = _find_categorical_variables(X[self.variables] if self.variables else X)
+        num_variables = _find_numerical_variables(X[self.variables] if self.variables else X)
+
+        data = pd.concat([X, y], axis=1) # column appending.
+        data.columns = list(X.columns) + ['target']
+
+        # split data into train and test sets
+        X_train, X_test, y_train, y_test = train_test_split(
+                                                data[cat_variables + num_variables + ['target']],
+                                                data['target'],
+                                                test_size = test_size,
+                                                random_state = random_state)
+
+        ## filling missing values.
+        for col in cat_variables:
+            X_train[col].fillna('M', inplace=True)
+            X_test[col].fillna('M', inplace=True)
+
+        for col in num_variables:
+            X_train[col].fillna(0, inplace=True)
+            X_test[col].fillna(0, inplace=True)
```

Comment:
it is the user's responsibility to make sure their variables do not contain missing values. Dataframes used with this transformer should be complete (that is, no missing values).

The categorical encoder and the discretiser would check that the data contains no na, and if it does, it would raise an error. So we are good to delete lines 131 to 138.